lib/dynamodb.js

```diff
@@ -1228,176 +1340,158 @@ DynamoDB.prototype.range = function (model, id, start, end, callback) {
-        }
-        break;
-      case 'between':
-        if (useIndex) {
-          addKeyCondition(key, "BETWEEN", conditionValue.between[0], conditionValue.between[1]);
+  let analyzeQueryWhere = (params, queryWhere) => {
+    let returnExpression = '';
+    let properties = Object.keys(queryWhere);
+    properties.forEach((key) => {
+      let conditionValue = queryWhere[key];
+      if (key === 'and') {
+        conditionValue.forEach((whereClause) => {
+          if (returnExpression !== '') {
+            returnExpression = returnExpression + ' AND ' + analyzeQueryWhere(params, whereClause);
+          } else {
+            returnExpression = analyzeQueryWhere(params, whereClause);
+          }
+        });
+      } else if (key === 'or') {
+        conditionValue.forEach((whereClause) => {
+          if (returnExpression !== '') {
+            returnExpression = returnExpression + ' OR ' + analyzeQueryWhere(params, whereClause);
+          } else {
+            returnExpression = analyzeQueryWhere(params, whereClause);
+          }
+        });
+      } else {
+        if (returnExpression !== '') {
+          // concadenate with ' AND '
+          returnExpression = returnExpression + ' AND ' + addExpressionToParam(params, key, conditionValue);
         } else {
-          prepFilterExpression();
-          params.FilterExpression += "(#" + key + " BETWEEN " + getExpressionValueToken(conditionValue.between[0]) + " AND " + getExpressionValueToken(conditionValue.between[1]) + ")";
+          returnExpression = addExpressionToParam(params, key, conditionValue);
         }
-        break;
-      case 'nin':
-        prepFilterExpression();
-        params.FilterExpression += "NOT " + "(#" + key + " IN (" + getExpressionValueToken(conditionValue.nin) + "))"
-        break;
-      case 'near':
-        throw new Error('near is not supported');
-        break;
-      case 'neq':
-        prepFilterExpression();
-        params.FilterExpression += "(#" + key + " <> " + getExpressionValueToken(conditionValue.neq) + ")";
-        break;
-      case 'like':
-        prepFilterExpression();
-        params.FilterExpression += "(contains(#" + key + "," + getExpressionValueToken(conditionValue.like) + "))";
-        break;
-      case 'nlike':
-        prepFilterExpression();
-        params.FilterExpression += "(NOT contains(#" + key + "," + getExpressionValueToken(conditionValue.nlike) + "))";
-        break;
-      case 'or':
-      case 'and':
-        throw new Error('and and or conditions are not supported');
       }
     });
-  } else {
-    //Simple case -- do equivalency in query/scan notation.
-    if (isIndexed(key, "eq", conditionValue, index)) {
-      addKeyCondition(key, "EQ", conditionValue);
-    } else {
-      prepFilterExpression();
-      params.FilterExpression += "#" + key + " = " + getExpressionValueToken(conditionValue);
-    }
+    return returnExpression;
```

Comment:
I feel like we should wrap the returnExpression in () because if you have

```
{
this: that,
that: something_else,
or: {
 or1: 1,
 or2: 2
}
}
```

You would end up with `this=that AND that=something_else AND or1=1 OR or2=2` and I can't remember what left-right associative means.